src/reanimated2/NativeReanimated/NativeReanimated.ts

```diff
@@ -62,6 +63,20 @@ export interface NativeReanimatedModule {
   ): void;
 }
 
+function checkReanimatedInstance() {
+  if (
+    global._REANIMATED_VERSION_JS !== undefined &&
+    global._REANIMATED_VERSION_JS !== jsVersion
+  ) {
+    throw new Error(
+      `[Reanimated] Another instance of \`react-native-reanimated\` was detected.
+See \`http://localhost:3000/react-native-reanimated/docs/guides/troubleshooting#reanimated-another-instance-of-react-native-reanimated-was-detected\` for more details. global._REANIMATED_VERSION_JS: ${global._REANIMATED_VERSION_JS}, jsVersion: ${jsVersion}}`
```

Comment:
I think there's no point in mentioning `global._REANIMATED_VERSION_JS` and `jsVersion` in the error message itself since it's an implementation detail.